### PR TITLE
Use cache accessor class when compiling reloc shaders

### PR DESCRIPTION
**Pull request by s-perron**
**Edit** Based on https://github.com/GPUOpen-Drivers/llpc/pull/1236

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
### Test summary for commit 85dcd994f9b649f0e8d0deb2f32fa16f54432aa4
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev 0168557a0715bf47a55ffe9858276cde1e74cc1d</li>
<li>SPVGEN: dev 5bf4618ac00d4675f57ca5b571a485bbc4d4fa33</li>
<li>XGL: dev d5d37d69a0d4bd01cc93b8e806b18f4040a13e6d</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 2d65a54ff6a0f09b40937c3efb03c82d42002187</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/224667)</summary><ul><li>Built with version 1.2.5.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 30577/56304 (54.3%)</li>
<li>Failed: 0/56304 (0.0%)</li>
<li>Not Supported: 25727/56304 (45.7%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 30577/56304 (54.3%)</li>
<li>Failed: 0/56304 (0.0%)</li>
<li>Not Supported: 25727/56304 (45.7%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 30319/55755 (54.4%)</li>
<li>Failed: 0/55755 (0.0%)</li>
<li>Not Supported: 25436/55755 (45.6%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 34150/56304 (60.7%)</li>
<li>Failed: 0/56304 (0.0%)</li>
<li>Not Supported: 22154/56304 (39.3%)</li>
</ul></details>


</ul></details>

**Review by amdrexu — COMMENTED** [review]
LGTM. I think there could be two PRs to make the change list clear.
1. A PR adding cache accessor interface.
2. A PR adding various user cases (should be based on PR 1)

File: llpc/util/llpcCacheAccessor.h [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ *  SOFTWARE.
+ *
+ **********************************************************************************************************************/
+/**
+ ***********************************************************************************************************************
+ * @file  llpcUtil.h
```
[the remark this answers is not in the document]
Comment (s-perron) [reply llpc/util/llpcCacheAccessor.h:27]: Good catch.  Cut and past, and I only fixed up the source file.

**Comment by amdvlk-admin** [comment]
### Test summary for commit 3f39cf4026c89bbf622c9bc872f6fc9df2c7a761
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev 0168557a0715bf47a55ffe9858276cde1e74cc1d</li>
<li>SPVGEN: dev 5bf4618ac00d4675f57ca5b571a485bbc4d4fa33</li>
<li>XGL: dev d5d37d69a0d4bd01cc93b8e806b18f4040a13e6d</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 2d65a54ff6a0f09b40937c3efb03c82d42002187</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/225198)</summary><ul><li>Built with version 1.2.5.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 30577/56304 (54.3%)</li>
<li>Failed: 0/56304 (0.0%)</li>
<li>Not Supported: 25727/56304 (45.7%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 30577/56304 (54.3%)</li>
<li>Failed: 0/56304 (0.0%)</li>
<li>Not Supported: 25727/56304 (45.7%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 30682/56286 (54.5%)</li>
<li>Failed: 0/56286 (0.0%)</li>
<li>Not Supported: 25604/56286 (45.5%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 34150/56304 (60.7%)</li>
<li>Failed: 0/56304 (0.0%)</li>
<li>Not Supported: 22154/56304 (39.3%)</li>
</ul></details>


</ul></details>

**Review by amdrexu — APPROVED** [review]

**Comment by JaxLinAMD** [comment]
retest this please

**Comment by amdvlk-admin** [comment]
### Test summary for commit 6e8584a633b1e5e631d86c8d22c8eeb7c669fa36
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev de5f376ca30a108f1fc1fd5cdf33fdfe9f9d5dbb</li>
<li>SPVGEN: dev 5bf4618ac00d4675f57ca5b571a485bbc4d4fa33</li>
<li>XGL: dev d81362ec6baf91a98895572d721ce7497efc8eeb</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev f4a2a23c8dd892c93384572bf0699ec3566e0e17</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/225198)</summary><ul><li>Built with version 1.2.5.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 30577/56304 (54.3%)</li>
<li>Failed: 0/56304 (0.0%)</li>
<li>Not Supported: 25727/56304 (45.7%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 30577/56304 (54.3%)</li>
<li>Failed: 0/56304 (0.0%)</li>
<li>Not Supported: 25727/56304 (45.7%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 30682/56286 (54.5%)</li>
<li>Failed: 0/56286 (0.0%)</li>
<li>Not Supported: 25604/56286 (45.5%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 34150/56304 (60.7%)</li>
<li>Failed: 0/56304 (0.0%)</li>
<li>Not Supported: 22154/56304 (39.3%)</li>
</ul></details>


</ul></details>

**Comment by trenouf** [comment]
retest this please

**Comment by trenouf** [comment]
Ah, it looks like that ret*st th*s ple*se like only retriggers the jenkins job, not the other tests. How do we retrigger the other tests? Does the PR owner have to push a new version? @JaxLinAMD

**Comment by trenouf** [comment]
Found it -- the "rerun jobs" button when you look at the details of one of the jobs.

**Comment by amdvlk-admin** [comment]
### Test summary for commit 6e8584a633b1e5e631d86c8d22c8eeb7c669fa36
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev de5f376ca30a108f1fc1fd5cdf33fdfe9f9d5dbb</li>
<li>SPVGEN: dev 5bf4618ac00d4675f57ca5b571a485bbc4d4fa33</li>
<li>XGL: dev d81362ec6baf91a98895572d721ce7497efc8eeb</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev f4a2a23c8dd892c93384572bf0699ec3566e0e17</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/225198)</summary><ul><li>Built with version 1.2.5.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 30577/56304 (54.3%)</li>
<li>Failed: 0/56304 (0.0%)</li>
<li>Not Supported: 25727/56304 (45.7%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 30577/56304 (54.3%)</li>
<li>Failed: 0/56304 (0.0%)</li>
<li>Not Supported: 25727/56304 (45.7%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 30682/56286 (54.5%)</li>
<li>Failed: 0/56286 (0.0%)</li>
<li>Not Supported: 25604/56286 (45.5%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 34150/56304 (60.7%)</li>
<li>Failed: 0/56304 (0.0%)</li>
<li>Not Supported: 22154/56304 (39.3%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit 060361955d5449d35de20d81c4f0ffc3c1560072
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev de5f376ca30a108f1fc1fd5cdf33fdfe9f9d5dbb</li>
<li>SPVGEN: dev 5bf4618ac00d4675f57ca5b571a485bbc4d4fa33</li>
<li>XGL: dev d81362ec6baf91a98895572d721ce7497efc8eeb</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 06c125a85ee8f4f787e6c8b977482ea2d1933ead</li>
</ul></details>

<details><summary>CTS tests (Failed: 1/181915)</summary><ul><li>Built with version 1.2.5.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 30577/56304 (54.3%)</li>
<li>Failed: 0/56304 (0.0%)</li>
<li>Not Supported: 25727/56304 (45.7%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 7669/13021 (58.9%)</li>
<li><details><summary>Failed: 1/13021 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.pipeline.vertex_input.multiple_attributes.binding_one_to_many.attributes_sequential.int.float.mat2
Stack trace: Script:
Crash

```
</p></details></li>

<li>Not Supported: 5351/13021 (41.1%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 30682/56286 (54.5%)</li>
<li>Failed: 0/56286 (0.0%)</li>
<li>Not Supported: 25604/56286 (45.5%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 34150/56304 (60.7%)</li>
<li>Failed: 0/56304 (0.0%)</li>
<li>Not Supported: 22154/56304 (39.3%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit 55e8862c6e7cebcf36c1555d78199e8562bd58ae
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev de5f376ca30a108f1fc1fd5cdf33fdfe9f9d5dbb</li>
<li>SPVGEN: dev 5bf4618ac00d4675f57ca5b571a485bbc4d4fa33</li>
<li>XGL: dev d81362ec6baf91a98895572d721ce7497efc8eeb</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 06c125a85ee8f4f787e6c8b977482ea2d1933ead</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/225198)</summary><ul><li>Built with version 1.2.5.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 30577/56304 (54.3%)</li>
<li>Failed: 0/56304 (0.0%)</li>
<li>Not Supported: 25727/56304 (45.7%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 30577/56304 (54.3%)</li>
<li>Failed: 0/56304 (0.0%)</li>
<li>Not Supported: 25727/56304 (45.7%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 30682/56286 (54.5%)</li>
<li>Failed: 0/56286 (0.0%)</li>
<li>Not Supported: 25604/56286 (45.5%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 34150/56304 (60.7%)</li>
<li>Failed: 0/56304 (0.0%)</li>
<li>Not Supported: 22154/56304 (39.3%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit 542b0d139acac637fde6b099237144088bc334d2
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev de5f376ca30a108f1fc1fd5cdf33fdfe9f9d5dbb</li>
<li>SPVGEN: dev 5bf4618ac00d4675f57ca5b571a485bbc4d4fa33</li>
<li>XGL: dev d81362ec6baf91a98895572d721ce7497efc8eeb</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 06c125a85ee8f4f787e6c8b977482ea2d1933ead</li>
</ul></details>

<details><summary>CTS tests (Failed: 1/196769)</summary><ul><li>Built with version 1.2.5.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 30577/56304 (54.3%)</li>
<li>Failed: 0/56304 (0.0%)</li>
<li>Not Supported: 25727/56304 (45.7%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 20544/27875 (73.7%)</li>
<li><details><summary>Failed: 1/27875 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.compute.basic.ssbo_unsized_arr_multiple_groups
Stack trace: Script:
Crash

```
</p></details></li>

<li>Not Supported: 7330/27875 (26.3%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 30682/56286 (54.5%)</li>
<li>Failed: 0/56286 (0.0%)</li>
<li>Not Supported: 25604/56286 (45.5%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 34150/56304 (60.7%)</li>
<li>Failed: 0/56304 (0.0%)</li>
<li>Not Supported: 22154/56304 (39.3%)</li>
</ul></details>


</ul></details>

**Comment by JaxLinAMD** [comment]
retest this please

**Comment by amdvlk-admin** [comment]
### Test summary for commit 542b0d139acac637fde6b099237144088bc334d2
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev de5f376ca30a108f1fc1fd5cdf33fdfe9f9d5dbb</li>
<li>SPVGEN: dev 5bf4618ac00d4675f57ca5b571a485bbc4d4fa33</li>
<li>XGL: dev d81362ec6baf91a98895572d721ce7497efc8eeb</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 06c125a85ee8f4f787e6c8b977482ea2d1933ead</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/226971)</summary><ul><li>Built with version 1.2.5.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 30987/56746 (54.6%)</li>
<li>Failed: 0/56746 (0.0%)</li>
<li>Not Supported: 25759/56746 (45.4%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 30985/56746 (54.6%)</li>
<li>Failed: 0/56746 (0.0%)</li>
<li>Not Supported: 25761/56746 (45.4%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 31140/56733 (54.9%)</li>
<li>Failed: 0/56733 (0.0%)</li>
<li>Not Supported: 25593/56733 (45.1%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 34575/56746 (60.9%)</li>
<li>Failed: 0/56746 (0.0%)</li>
<li>Not Supported: 22171/56746 (39.1%)</li>
</ul></details>


</ul></details>

**Pull request merged**